partiql-planner/src/main/kotlin/org/partiql/planner/internal/typer/PlanTyper.kt

```diff
@@ -621,71 +625,29 @@ internal class PlanTyper(
         private fun toRexCall(
             match: FnMatch.Ok<FunctionSignature.Scalar>,
             args: List<Rex>,
-            isNotMissable: Boolean,
         ): Rex {
             // Found a match!
             val newFn = fnResolved(match.signature)
             val newArgs = rewriteFnArgs(match.mapping, args)
-            val returns = newFn.signature.returns
 
-            // 7.1 All functions return MISSING when one of their inputs is MISSING (except `=`)
-            newArgs.forEach {
-                if (it.type == MissingType && !isNotMissable) {
-                    handleAlwaysMissing()
-                    return rex(MISSING, rexOpCallStatic(newFn, newArgs))
-                }
+            // Check literal missing inputs
+            val argAlwaysMissing = args.any {
+                val op = it.op as? Rex.Op.Lit ?: return@any false
+                op.value.type == PartiQLValueType.MISSING
             }
-
-            // If a function is NOT Missable (i.e., does not propagate MISSING)
-            // then treat MISSING as null.
-            var isMissing = false
-            var isMissable = false
-            if (isNotMissable) {
-                if (newArgs.any { it.type is MissingType }) {
-                    isMissing = true
-                } else if (newArgs.any { it.type.isMissable() }) {
-                    isMissable = true
-                }
-            }
-
-            // Determine the nullability of the return type
-            var isNull = false // True iff NULL CALL and has a NULL arg
-            var isNullable = false // True iff NULL CALL and has a NULLABLE arg; or is a NULLABLE operator
-            if (newFn.signature.isNullCall) {
-                if (isMissing) {
-                    isNull = true
-                } else if (isMissable) {
-                    isNullable = true
-                } else {
-                    for (arg in newArgs) {
-                        if (arg.type is NullType) {
-                            isNull = true
-                            break
-                        }
-                        if (arg.type.isNullable()) {
-                            isNullable = true
-                            break
-                        }
-                    }
+            if (argAlwaysMissing) {
+                // TODO: The V1 branch has support for isMissable and isMissingCall. This codebase, however, does not
+                //  have support for these concepts yet. This specific commit (see Git blame) does not seek to add this
+                //  functionality. Below is a work-around for the lack of "isMissable" and "isMissingCall"
+                if (match.signature.name !in listOf("is_null", "is_missing", "eq")) {
```

Comment:
Do we also need to include the logical ops here (and, or, not)? When `AND` and `OR` are provided with a `NULL`/`MISSING` argument, they may still return a boolean:

```
PartiQL> NULL AND false;
==='
false
---
OK!
PartiQL> NULL OR true;
==='
true
---
OK!
```